opensquirrel/passes/merger/general_merger.py

```diff
@@ -71,14 +114,17 @@ def try_name_anonymous_bloch(bsr: BlochSphereRotation) -> BlochSphereRotation:
          A default BlockSphereRotation if this BlochSphereRotation is close to it,
          or the input BlochSphereRotation otherwise.
     """
-    for gate_function in default_bloch_sphere_rotation_without_params_set.values():
-        gate = gate_function(*bsr.get_qubit_operands())
-        if (
-            np.allclose(gate.axis, bsr.axis, atol=ATOL)
-            and np.allclose(gate.angle, bsr.angle, atol=ATOL)
-            and np.allclose(gate.phase, bsr.phase, atol=ATOL)
-        ):
-            return gate
+    if not bsr.is_anonymous:
+        return bsr
+    for default_bsr_callable in default_bloch_sphere_rotation_set.values():
+        if can_invoke_bsr_callable(default_bsr_callable, bsr):
+            default_bsr = default_bsr_callable(*bsr.get_qubit_operands())
+            if (
+                np.allclose(default_bsr.axis, bsr.axis, atol=ATOL)
+                and np.allclose(default_bsr.angle, bsr.angle, atol=ATOL)
+                and np.allclose(default_bsr.phase, bsr.phase, atol=ATOL)
```

Comment:
These are just a single value, so no need to use `np.allclose()`
```suggestion
                and abs(default_bsr.angle - bsr.angle) < ATOL
                and abs(default_bsr.phase - bsr.phase) < ATOL
```